cmake/compile_tests/pthread.cpp

```diff
@@ -1,14 +0,0 @@
-#include <pthread.h>
-
-void* kokkos_test(void* args) { return args; }
```

Comment:
Oversight forgot to remove that file in 9147621eb30f6beb3fd0836db7b86bd77d265d36